Save transients after a migration has been completed

Ferdev opened the pull request:
-------

## Description
When the WooCommerce Shipping & Tax migration to WooCommerce Shipping and WooCommerce Tax is completed, we need to display migration status messages for each entry of these plugins in the plugins list screen. One issue is the migration status is stored in the WooCommerce Shipping & Tax options object, and the class helper that access that options object is not available for WooCommerce Shipping and WooCommerce Tax as WooCommerce Shipping & Tax is disabled. This PR solves that issue by creating a couple of transients that tell WooCommerce Shipping and WooCommerce Tax that the migration has been completed successfully. The transients last one day until they're deleted.

### Related issue(s)

N/A

### Steps to reproduce & screenshots/GIFs

1. Checkout this branch and make sure you're eligible to run the migration:

	- Reset the `wcshipping_migration_state` option with `WC_Connect_Options::delete_option( 'wcshipping_migration_state' );`
	- Use the WooCommerce Connect Server staging endpoint: `https://api-staging.woocommerce.com`

2. Go to the orders list, and click the migration button at the top of the screen. Wait for the migration to finish.

![shot-2024-07-06-13-51-19](https://github.com/Automattic/woocommerce-services/assets/130142/39186f35-60fb-48c8-9899-feb5b3284bb5)


3. Install a transients manager plugin and check if the transients `wcshipping_migration_completed` and `wctax_migration_completed` have been created.

### Checklist

<!-- All testable code should have tests. -->
- [ ] unit tests

<!-- An entry in the changelog file is always required -->
- [ ] `changelog.txt` entry added
- [ ] `readme.txt` entry added

Ferdev committed:
Save transients after a migration has been completed

harriswong commented:
> One issue is the migration status is stored in the WooCommerce Shipping & Tax options object, and the class helper that access that options object is not available for WooCommerce Shipping and WooCommerce Tax as WooCommerce Shipping & Tax is disabled.

For `wcshipping_migration_state`, would it be better if we change all code:
`WC_Connect_Options::get_option()` --> `get_option()`
`WC_Connect_Options::update_option()` --> `update_option()` 

This way, we decouple the WCS&T class helper from the option itself, making the option available to anything that has access to `wp_options` table.

Ferdev committed:
Use general options instead of WooCommerce Shipping & Tax's options

Ferdev commented:
Yeah, good idea Harris! I just pushed a new commit with that implementation

harriswong commented on woocommerce-services.php:
```diff
@@ -265,9 +265,9 @@ public static function plugin_deactivation() {
+			$migration_state = get_option( 'wcshipping_migration_state' );
 			if ( $migration_state !== WC_Connect_WCST_To_WCShipping_Migration_State_Enum::COMPLETED ) {
-				WC_Connect_Options::update_option( 'wcshipping_migration_state', WC_Connect_WCST_To_WCShipping_Migration_State_Enum::COMPLETED );
+				update_option( 'wcshipping_migration_state', WC_Connect_WCST_To_WCShipping_Migration_State_Enum::COMPLETED );
```
The `get_option()` doesn't type cast and `"12 !== 12`. After the migration is done, the banner will continue to show. 
![image](https://github.com/Automattic/woocommerce-services/assets/572862/86818031-f60f-42b3-9c4b-75de0d01c8b9)

harriswong reviewed (changes requested):
I tested the "Update now" flow 
![image](https://github.com/Automattic/woocommerce-services/assets/572862/7cc79032-c331-4328-8b76-31ab1487f22a), but the banner kept showing, i think we need to typecast. 

Also, `wordpress/wp-content/plugins/woocommerce-services/classes/class-wc-connect-options.php` line 55 `'wcshipping_migration_state',` isn't needed anymore, we can remove. 

Thanks!

Ferdev committed:
Cast the migration state to int now that is saved as a global option

Ferdev committed:
Remove wcshipping_migration_state as option name

Ferdev commented:
:wave: I've updated the code to cast o int all migration state ocurrences wherever they're read from the database. I'd forgotten that scalar values are returned as strings by `get_option`! What I wonder is why I didn't have issues with the popup not hidding... :thinking: Anyway, just updated the code, let me know what you think!

harriswong reviewed (approved):
There is a conflict with `wcs-migration`, but code LGTM. I tested the modal migration from the order listing page and the individual order page 👍 

`WC_Connect_Options` serialized the option so we didn't need to check for type. But the WP option one is straight from the DB table so we do need to check for type. Instead of doing type-safe comparison, `!==`, I was thinking maybe we can do `!=` instead. Then, we don't need all these `intval()` everywhere. The condition for `$migration_pending` is to return a boolean so I think using a typeless comparison would be fine too. 

This is good to merge once the merge conflict is resolved and if the migration button still works after the merge conflict!

Ferdev committed:
Merge remote-tracking branch 'origin/wcs-migration' into feature/save-transients-migration-completed
